src/codec.rs

```diff
@@ -16,51 +16,73 @@
 
 use crate::alloc::vec::Vec;
 use crate::alloc::boxed::Box;
+use crate::alloc::string::{String, ToString};
 use crate::alloc::collections::btree_map::BTreeMap;
 
+
 #[cfg(any(feature = "std", feature = "full"))]
 use crate::alloc::{
-	string::String,
 	borrow::{Cow, ToOwned},
 };
 
 use core::{mem, slice};
 use arrayvec::ArrayVec;
 use core::marker::PhantomData;
 
+#[cfg_attr(feature = "std", derive(Debug))]
+#[derive(PartialEq)]
+pub struct Error(pub String);
```

Comment:
Using a `String` as internal representation is maybe not the best idea. On `no_std` we will get into trouble with this.
Could we switch to `&'static str`?

Reply:
yeah, updated the PR